crates/engine/schema/src/builder/coerce/extension.rs

```diff
@@ -0,0 +1,329 @@
+use crate::{
+    builder::{extension::ExtensionSdl, GraphContext},
+    extension::{ExtensionInputValueRecord, InjectionStage},
+};
+use cynic_parser::{
+    common::{TypeWrappersIter, WrappingType},
+    type_system::{
+        Definition, EnumDefinition, InputObjectDefinition, InputValueDefinition, ScalarDefinition, Type, TypeDefinition,
+    },
+    ConstValue,
+};
+use federated_graph::Value;
+
+use super::{value_path_to_string, ExtensionInputValueError, InputValueError};
+
+pub(crate) struct ExtensionInputValueCoercer<'a, 'b> {
```

Comment:
Pretty much a copy-pasted coercer but adapted to the relevant types + extra validations as we don't assume types necessarily exists.